alg/sha256.c

```diff
@@ -397,13 +397,17 @@ SHA256_Init(SHA256_CTX * ctx)
 	/* Initialize state. */
 	memcpy(ctx->state, initial_state, sizeof(initial_state));
 
-	/* Choose hardware acceleration (if any). */
-	if (hwaccel == HW_UNSET) {
-		hwaccel_init();
+#ifdef HWACCEL
+	/* We're finished if we've set hwaccel. */
+	if (hwaccel != HW_UNSET)
```

Comment:
Sorry, I wasn't clear.  I meant putting this bit into `hwaccel_init`.  So `SHA256_Init` always calls `hwaccel_init` but `hwaccel_init` does nothing if it has already initialized.